src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1479,11 +1481,15 @@ ExecuteModifyTasks(List *taskList, bool expectResults, ParamListInfo paramListIn
 	 * ProcessUtility, so we only need to do this for DML commands.
 	 */
 	firstTask = (Task *) linitial(taskList);
-	firstShardInterval = LoadShardInterval(firstTask->anchorShardId);
-	if (PartitionedTable(firstShardInterval->relationId) &&
-		firstTask->taskType == MODIFY_TASK)
+	if (firstTask->taskType == MODIFY_TASK)
```

Comment:
Do we really hit these lines? I guess this is done before we decide  to disable router executor for type creation?

Reply:
I hit this case with a `DDLJob` that has tasks that are not anchored to a shard.

By having the task type check earlier we prevent calling `LoadShardInterval` with `INVALID_SHARD_ID`.